src/core/components/config.js

```diff
@@ -238,7 +234,9 @@ export default class {
   }
 
   setUUID(val: string): this {
-    if (this._db && this._db.set) this._db.set(`${this.subscribeKey}uuid`, val);
+    if (!val || typeof val !== 'string' || val.trim().length === 0) {
+      throw new Error('Missing uuid parameter. Provide a valid string uuid');
+    }
     this.UUID = val;
```

Comment:
Looks good, I know we discussed this and I will hold back my concern on throwing exceptions from constructors :) it is a C# and Java thing (something to do with destructor cleanup of resources that is not relevant here) which made me take that stand. For this it should be fine.

Reply:
since constructor calls setUUID it will have potential to throw when constructor is called